src/generators/typescript/TypeScriptFileGenerator.ts

```diff
@@ -0,0 +1,24 @@
+import { TypeScriptGenerator, TypeScriptRenderCompleteModelOptions } from './';
+import { CommonInputModel, OutputModel } from '../../models';
+import * as path from 'path';
+import { AbstractFileGenerator } from '../AbstractFileGenerator';
+import { FileHelpers } from '../../helpers';
+
+export class TypeScriptFileGenerator extends TypeScriptGenerator implements AbstractFileGenerator<TypeScriptRenderCompleteModelOptions> {
+  /**
+   * Generates all the models to an output directory each model with their own separate files. 
+   * 
+   * @param input
+   * @param outputDirectory where you want the models generated to
+   * @param options
+   */
+  public async generateToFiles(input: Record<string, unknown> | CommonInputModel, outputDirectory: string, options?: TypeScriptRenderCompleteModelOptions): Promise<OutputModel[]> {
```

Comment:
I wrote about that in PR related to the JS File Renderer, but I'll write about that also here.

Have in mind that someone will want to render models with ESM not CJS module system, so you `TypeScriptRenderCompleteModelOptions` should have option to choose it - by default we can use ESM.

Reply:
Yep, changing 👍